src/src/hw9/hillel/ArrayValueCalculator.java

```diff
@@ -0,0 +1,40 @@
+package hw9.hillel;
+
+public class ArrayValueCalculator {
+
+    public static int doCalc(String[][] arr) throws ArraySizeException, ArrayDataException {
+        if (arr.length != 4 || arr[0].length != 4) {
+            throw new ArraySizeException("Array size must be 4x4.");
+        }
+
+        int sum = 0;
+        for (int i = 0; i < arr.length; i++) {
+            for (int j = 0; j < arr[0].length; j++) {
```

Comment:
Oops, it seems that you have a mistake here. You have `j < arr[0].length;` meaning that you're looping over the same row all the time! It should be like this: `j < arr[i].length;`